src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -67,16 +162,74 @@ const DesignColors = () => {
 				} );
 			}
 			setSelectedColors( colorsArray );
+			//  Save selected palette color to Store
 			currentData.data.colorStyle = colorStyle;
 			setCurrentOnboardingData( currentData );
 			return colorsArray;
 		}
 	}
 
+	/**
+	 * When the user clicks a predefined palette or reset then
+	 * The custom colors should be reset to the original colors.
+	 */
+	function clearCustomColors() {
+		for ( const custom in customColors ) {
+			customColors[ custom ] = '';
+		}
+		// Resetting the color palette to default and unsetting the selected predefined color palette, if any.
+		setCustomColors( customColors );
+	}
+
+	/**
+	 * When the user clicks on reset it fetches the
+	 * orginal colors and replaces them in the preview
+	 */
+	async function setToDefaultPalette() {
+		const globalStyles = await getGlobalStyles( true );
+		let selectedGlobalStyle;
+		if ( currentData?.data?.theme?.variation ) {
+			selectedGlobalStyle = globalStyles.body.filter(
+				( globalStyle ) =>
+					globalStyle.title === currentData.data.theme.variation
+			)[ 0 ];
+		} else if ( globalStyles.body[ 0 ]?.id === 0 ) {
+			selectedGlobalStyle = globalStyles.body[ 0 ];
+		}
+		updatePreviewSettings(
+			// eslint-disable-next-line react-hooks/rules-of-hooks
+			useGlobalStylesOutput( selectedGlobalStyle, storedPreviewSettings )
+		);
+
+		clearCustomColors();
+
+		const selectedGlobalStylePalette =
+			selectedGlobalStyle.settings.color.palette;
+		setSelectedColors( selectedGlobalStylePalette );
+		setSelectedColorsLocal( stateToLocal( selectedGlobalStylePalette ) );
+		trackHiiveEvent( 'color-selection-reset', selectedGlobalStyle.title );
+	}
+
+	/**
+	 * Reset Button to reset selected colors
+	 */
+	async function resetColors() {
+		setToDefaultPalette();
+		currentData.data.colorStyle = '';
+		setCurrentOnboardingData( currentData );
+	}
+
+	/**
+	 * Converts the user selected value into a suitable valid global styles array value
+	 *
+	 * @param {string} colorStyle        - Selected Color Palette slug
+	 * @param {Object} colorPalettesTemp - Color Palette from Backend with colors for every palette
+	 * @param {Object} globalStylesTemp  - Global Styles from the store
+	 * @return {Object} selectedGlobalStyle - Updated Global Styles with new color changes
+	 */
 	async function saveThemeColorPalette(
 		colorStyle,
 		colorPalettesTemp = colorPalettes,
-		selectedColorsLocalTemp = selectedColors,
 		globalStylesTemp = storedPreviewSettings
 	) {
```

Comment:
This one can still be simplified.
1.  Remove all the `temp` variable usage.
2. Change `colorPalettesTemp && colorStyle && selectedThemeColorPalette` to a negation and return.
3. Switch to using `map` instead of modifying the same variable that is being iterated on.